src/Adapter/Country/CountryRequiredFieldsProvider.php

```diff
@@ -0,0 +1,86 @@
+<?php
+/**
+ * 2007-2019 PrestaShop SA and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Adapter\Country;
+
+use Country;
+use PrestaShop\PrestaShop\Core\Country\CountryRequiredFieldsProviderInterface;
+use PrestaShop\PrestaShop\Core\Domain\Country\Exception\CountryNotFoundException;
+use PrestaShop\PrestaShop\Core\Domain\Country\ValueObject\CountryId;
+use PrestaShopException;
+
+class CountryRequiredFieldsProvider implements CountryRequiredFieldsProviderInterface
+{
+    /**
+     * {@inheritdoc}
+     *
+     * @throws CountryNotFoundException
+     */
+    public function isStatesRequired(CountryId $countryId): bool
+    {
+        return $this->getCountry($countryId)->contains_states;
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * @throws CountryNotFoundException
+     */
+    public function isDniRequired(CountryId $countryId): bool
+    {
+        return $this->getCountry($countryId)->need_identification_number;
+    }
+
+    /**
+     * Get legacy country
+     *
+     * @param CountryId $countryId
+     *
+     * @return Country
+     *
+     * @throws CountryNotFoundException
+     */
+    private function getCountry(CountryId $countryId): Country
+    {
+        $countryIdValue = $countryId->getValue();
+
+        try {
+            $country = new Country($countryIdValue);
+        } catch (PrestaShopException $e) {
+            throw new CountryNotFoundException(
+                sprintf('Country with id "%s" was not found.', $countryIdValue)
```

Comment:
Could be another exception, the message must be more generic or use the `$e->getMessage()`. 
Otherwise you can do this to prevent duplicate messages.

```php
try {
    $country = new Country($countryIdValue);
} catch (PrestaShopException $e) {
}

if (empty($country->id) || $country->id !== $countryIdValue) {
    throw new CountryNotFoundException(
        sprintf('Country with id "%s" was not found.', $countryIdValue)
    );
}
```